main.tf

```diff
@@ -59,61 +61,44 @@ resource "aws_route_table" "public" {
   }
 }
 
-resource "aws_route_table_association" "public1" {
-  subnet_id      = aws_subnet.public1.id
-  route_table_id = aws_route_table.public.id
-}
+resource "aws_route_table_association" "public" {
+  count = local.public_RT_ass
```

Comment:
use `count = length(local.public_cidr)`for resources that are more than 1. 
e.g replace `count = local.public_RT_ass` with `count = length(local.public_cidr)` since both will return the value 2